src/EFCore.PG/Query/Internal/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -246,6 +253,52 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCall)
                 }
             }
 
+            {
+                // Pattern match for: array.Any(e => e == x) (and other equality patterns)
+                // Transform this to Contains.
+                if (method.IsClosedFormOf(EnumerableAnyWithPredicate) &&
+                    arguments[1] is LambdaExpression wherePredicate &&
+                    TryMatchEquality(wherePredicate.Body, out var left, out var right) &&
+                    (left == wherePredicate.Parameters[0] || right == wherePredicate.Parameters[0]))
+                {
+                    var item = left == wherePredicate.Parameters[0]
+                        ? right
+                        : right == wherePredicate.Parameters[0]
+                            ? left
+                            : null;
+
+                    return item is null
+                        ? null
+                        : Visit(Expression.Call(Contains.MakeGenericMethod(method.GetGenericArguments()[0]), array, item));
+                }
+
+                static bool TryMatchEquality(Expression expression, out Expression left, out Expression right)
+                {
+                    if (expression is BinaryExpression binary)
+                    {
+                        (left, right) = (binary.Left, binary.Right);
```

Comment:
While I use tuple assignments often, I don't think they looks well outside of expression bodied methods.

Reply:
Embrace the Python style :)

Reply:
Let's leave two line instead of one, really hard to read. If there is a comma operator allowing to write a single line instead of three, I will be happy.

Reply:
Even the conservative guys in EF have agreed to me doing this... Seriously though - I know it's new syntax, it just requires getting used to...